packages/@lwc/engine-dom/src/patches/global-registry.ts

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { setPrototypeOf, isUndefined, defineProperties } from '@lwc/shared';
+
+const { HTMLElement: NativeHTMLElement } = window;
+const {
+    hasAttribute: nativeHasAttribute,
+    setAttribute: nativeSetAttribute,
+    removeAttribute: nativeRemoveAttribute,
+    getAttribute: nativeGetAttribute,
+} = NativeHTMLElement.prototype;
+
+const definitionForElement = new WeakMap<HTMLElement, Definition>();
+const pendingRegistryForElement = new WeakMap<HTMLElement, Definition>();
+const definitionForConstructor = new WeakMap<CustomElementConstructor, Definition>();
+
+const pivotCtorByTag = new Map<string, CustomElementConstructor>();
+const definitionsByTag = new Map<string, Definition>();
```

Comment:
this should probably be renamed to `globalDefinitionsByTag` to highlight that this is the new global registry.